lxd/patches.go

```diff
@@ -1335,4 +1338,154 @@ func patchRemoveCandidRBACConfigKeys(_ string, d *Daemon) error {
 	return nil
 }
 
+// patchStorageSetVolumeUUID sets a unique volatile.uuid field for each volume and its snapshots.
+func patchStorageSetVolumeUUID(_ string, d *Daemon) error {
+	s := d.State()
+
+	// Get all storage pool names.
+	pools, err := s.DB.Cluster.GetStoragePoolNames()
+	if err != nil {
+		// Skip the rest of the patch if no storage pools were found.
+		if api.StatusErrorCheck(err, http.StatusNotFound) {
+			return nil
+		}
+
+		return fmt.Errorf("Failed getting storage pool names: %w", err)
+	}
+
+	// Check if this member is the current cluster leader.
+	isLeader := false
+
+	if !d.serverClustered {
+		// If we're not clustered, we're the leader.
+		isLeader = true
+	} else {
+		leaderAddress, err := d.gateway.LeaderAddress()
+		if err != nil {
+			return err
+		}
+
+		if s.LocalConfig.ClusterAddress() == leaderAddress {
+			isLeader = true
+		}
+	}
+
+	poolIDNameMap := make(map[int64]string, 0)
+	poolVolumes := make(map[int64][]*db.StorageVolume, 0)
+	poolBuckets := make(map[int64][]*db.StorageBucket, 0)
+
+	for _, pool := range pools {
+		// Load storage pool.
+		loadedPool, err := storagePools.LoadByName(s, pool)
+		if err != nil {
+			return fmt.Errorf("Failed loading pool %q: %w", pool, err)
+		}
+
+		// Ensure the renaming is done only on the cluster leader for remote storage pools.
+		if loadedPool.Driver().Info().Remote && !isLeader {
```

Comment:
As this is only affecting the DB, I think we can do this all on the leader right?

Reply:
I dont mind btw, just a thought.

Reply:
I like this, we can make the patch simpler by removing `loadedPool`. 

I guess it is not possible to have LXD local pools in a clustered setup?

Reply:
well you can have local storage pools on every member of a clustered setup, but the pool record covers all cluster members.